### PR TITLE
add feature: colored tags

### PR DESCRIPTION
**Pull request by elfman**
## Description

for #2260.
new feature: colored tags, user can customize the color of tags

## Type of changes

- :white_circle: Bug fix (Change that fixed an issue)
- :white_circle: Breaking change (Change that can cause existing functionality to change)
- :white_circle: Improvement (Change that improves the code. Maybe performance or development improvement)
- :radio_button: Feature (Change that adds new functionality)
- :white_circle: Documentation change (Change that modifies documentation. Maybe typo fixes)

## Checklist:

- :radio_button: My code follows [the project code style](docs/code_style.md)
- :white_circle: I have written test for my code and it has been tested
- :radio_button: All existing tests have been passed
- :radio_button: I have attached a screenshot/video to visualize my change if possible

### CODE REVIEW

File: browser/components/NoteItem.js [hunk 1]
```diff
@@ -27,15 +27,15 @@ const TagElement = ({ tagName }) => (
...
  * @return {React.Component}
  */
-const TagElementList = (tags, showTagsAlphabetically) => {
+const TagElementList = (tags, showTagsAlphabetically, tagConfig) => {
   if (!isArray(tags)) {
     return []
   }
 
   if (showTagsAlphabetically) {
     return _.sortBy(tags).map(tag => TagElement({ tagName: tag }))
```
Comment (daiyam) [inline browser/components/NoteItem.js]: `color: tagConfig[tag]` is missing

File: package.json [hunk 2]
```diff
@@ -97,6 +97,7 @@
     "react": "^15.5.4",
     "react-autosuggest": "^9.4.0",
     "react-codemirror": "^0.3.0",
+    "react-color": "^2.17.0",
```
Comment (daiyam) [inline package.json]: already in `devDependencies`

**Comment by daiyam** [comment]
@elfman could you use the PR template 😉 

Also could you rename your dictionary `config.tag` with a more explicit name and update the comment of modified functions.

File: browser/components/NoteItem.js (hunk 1 above)
Comment (elfman) [reply browser/components/NoteItem.js]: do you mean a runtime error?

**Comment by ZeroX-DG** [comment]
How about something like this. (A quick draw using figma)
frame 2

**Comment by ZeroX-DG** [comment]
@elfman The tag textColor doesn't feel right in the tag. You can use this module: https://github.com/onury/invert-color to generate the tag text color base on the tag background.

**Comment by elfman** [comment]
@ZeroX-DG Updated

**Comment by ZeroX-DG** [comment]
Looks nice! But the `x` symbol is....you get the idea :smile:
image

**Comment by elfman** [comment]
@ZeroX-DG Updated

**Comment by daiyam** [comment]
The line 36 of `browser/components/NoteItem.js` has not been fixed, yet 😉

**Comment by ZeroX-DG** [comment]
How careless of me! Sorry guys :cry:

**Comment by elfman** [comment]
@daiyam @ZeroX-DG fixed, please review again

File: browser/components/ColorPicker.styl [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+.btn-cancel,
+.btn-confirm,
+.btn-reset
+  vertical-align middle
+  height 25px
+  margin-top 2.5px
+  colorDefaultButton()
+  border-radius 2px
+  border $ui-border
```
Comment (ZeroX-DG) [inline browser/components/ColorPicker.styl]: It looks better with `border: none`. Also when I hover the button, its' background is transparent
ab

**Comment by daiyam** [comment]
@elfman it's ok for me

File: browser/components/ColorPicker.styl (hunk 3 above)
Comment (elfman) [reply browser/components/ColorPicker.styl]: @ZeroX-DG Updated

**Pull request merged**